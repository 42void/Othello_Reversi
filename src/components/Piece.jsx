import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Piece extends Component {
  constructor(props) {
    super(props);
    this.state = { };
  }

  render() {
    const { color } = this.props;
    return (
      <div style={{ backgroundColor: color === 1 ? '#000' : '#FFF' }} className="piece" />
    );
  }
}

Piece.propTypes = {
  color: PropTypes.number.isRequired,
};

export default Piece;
