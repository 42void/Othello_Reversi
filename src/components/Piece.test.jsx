import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Piece from './Piece';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
beforeEach(() => {
  wrapper = mount(<Piece color={1} />);
});

describe('Piece component', () => {
  it('renders', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Piece color={1} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('matches snapshot', () => {
    const component = renderer.create(<Piece color={1} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
