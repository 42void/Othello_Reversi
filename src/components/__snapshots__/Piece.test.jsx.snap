// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Piece component matches snapshot 1`] = `
<div
  className="piece"
  style={
    Object {
      "backgroundColor": "#000",
    }
  }
/>
`;
